components/server/src/config/config-inferrer.ts

```diff
@@ -41,29 +41,33 @@ export class ConfigInferrer {
         if (!pckjsonContent) {
             return;
         }
-        let command: "yarn" | "npm" = "npm";
+        let pckjson;
+        try {
+            pckjson = JSON.parse(pckjsonContent);
+        } catch (e) {
+            console.log(e, pckjsonContent);
+        }
+        let command: "yarn" | "npm" | "pnpm" = "npm";
```

Comment:
Is it worth extracting a type for this union now that it has 3 members?

Reply:
Since this is used in a single place, I think it's okay to leave it like that for now.